<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Genome Size Evolution Models by TSL</title>
    <meta name="viewport" content="width=device-width, initial-scale=1"/>
    <meta name="description" content="genome size evolution models in host-pathogen systems, increase in length and number of effector genes and transposable elements"/>
    <meta name"author" content="Alice Minotto"/>
    <meta property="og:title" content="Genome Size Evolution Models"/>
    <meta property="og:type" content="website"/>
    <meta property="og:image" content="http://github.com/Host-pathogen-evol/TSL-PROJECT-A/blob/gh-pages/images/6bakground_fb.png?raw=true"/>
    <meta property="og:description" content="Genome size evolution models in host-pathogen systems,
    increase in length and number of effector genes and transposable elements. Two models
    based on oomycetes and plant interactions studies show increase in size and host jump response"/>
    <meta name="twitter:description" content="Genome size evolution models in host-pathogen systems,
    increase in length and number of effector genes and transposable elements. Two models
    based on oomycetes and plant interactions studies">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">Genome Length Evolution in Host-Pathogen systems</h1>
      <h2 class="project-tagline">HOME OF  THE GENOME SIZE EVOLUTION PROJECT</h2>
      <h3 class="project-tagline">Sainsbury Laboratory, Norwich</h3>
      <a href="https://github.com/Host-pathogen-evol/TSL-PROJECT-A" class="btn">View on GitHub</a>
      <a href="https://github.com/Host-pathogen-evol/TSL-PROJECT-A/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/Host-pathogen-evol/TSL-PROJECT-A/tarball/master" class="btn">Download .tar.gz</a>
    </section>
    <section class="nav nav-pills" id="navigator">
        <a href="#first_model" class="btn2">First Model</a>
        <a href="#second_model" class="btn2">Second Model</a>
    </section>
    <section class="main-content">
      <h1>
<a id="genome-length-evolution-in-host-pathogen-systems" class="anchor" href="#genome-length-evolution-in-host-pathogen-systems" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Genome Length Evolution in Host-Pathogen systems.</h1>

<p>!!!!This page is currently under construction!!!!</p>

<p>This page contains a simple presentations of the models and the documentation for python and Ipython modules written for the
host-pathogen models detailed in <a href="http://calugo.github.io/blog/2014/10/14/genome-evolution-in-host-pathogen-systems/">Carlos Lugo's website</a>,
as well as a brief description of our results.</p>

<p>The project was born in <a href="http://www.tsl.ac.uk/groups/bioinformatics/">TSL Bioinformatics Group</a>, as a mean to model the evolution of plant pathogenâ€™s genomes,
    in particoular oomycetes' as <em>Phytophthora infestans</em>.</p>
<p>We developed two different mathematical-computational models to explore the changes in genome size and number of effector genes and transposable elements.
    One of the main interests was to investigate the effects of TEs on genome size, that is observed as increasing in oomycetes.
    The other purpose of the study was to model both the population dynamics and the genome changes after host jumps.</p>

<p>Do not hesitate to <a href="mailto:kupkasmale@gmail.com?subject=genome%20size%20evolution%20project">contact me</a> if you feel you need any more help or informations, or if you want to discuss the project.</p>
<p>Cheers!</p>

<hr color=rgba(0,0,0,0.5)/>

<a name="first_model"></a><h2><bold>First Model</bold></h2>

<p>In the first model used for the simulations, we are considering effector genes and transposable elements.</p>
<p>We considered the following events that can happen during evolution:
    <ul>
        <li>gain of effector genes or TEs by horizontal gene transfer,</li>
        <li>loss of effector genes or TEs,</li>
        <li>duplication of effector genes or TEs,</li>
        <li>mutation or recombination of effectors,</li>
        <li>silencing of effectors due to TE insertion.</li>
    </ul>
</p>
<p>Here each effector is characterized by a score (used to calculate the overall fitness) and a length.</p>
<p>In this very simple model hosts are no explicitly considered, and a host jump simply results in random changes
in the scores of the effectors in the pool.</p>
<p>We run the simulation both with and without jumps (you can find here
the code to perform the complete simulation with jumps, to skip this step just set the time gap between jumps
to the time of your simulation and number of jumps to <code>1</code>). Please note that we talk about time as a measure of evolutionary events,
not absolute time.</p>

<h3>Discussion</h3>

<p></p> <!--some results and observations-->

<h3>Running simulations and getting your own results</h3>

<p>To run a simulation of the first model you can download the codes <a href="https://github.com/Host-pathogen-evol/TSL-PROJECT-A/tree/master/FIRST_MODEL">here</a>.</p>
<p>There are a lot of parameters you may want to change, you can have a full list of them by running</p>
<code>$python mprostest.py -h</code>
<p>Plese note that you will need to previously install <a href="http://sourceforge.net/projects/pygsl/">pygsl</a> to run the simulation.</p>
<p>Also, we suggest to seed the random number generator with the <code>-s</code> or <code>--seed</code> option, as the program is using a default seed.</p>
<p>It is advisable to set the path to a new folder not to overwrite previous results, and because the program could produce a large
    amount of folders.</p>

<hr color=rgba(0,0,0,0.5)/>

<a name="second_model"></a><h2><bold>Second Model</bold></h2>

<p>In the second model we decided not to explicitly consider transposable elements anymore, as their evolution pattern resulted quite clear from the
first model, and there are no other known biological effects today to add in the model. In particoular it is not proved that they have an effect on the pathogen fitness,
other than contributing to silencing or general genome plasticity, both of which can be implicitly considered modifying accordingly
the rates of mutation/recombination, silencing, hgt and duplication.</p>
<p>In this model effectors are characterized by a length and a list of target genes, each one with a score.</p>
<p>Moreover the host is characterized by a fixed list of target.</p>
<p>The fitness is so calculated considering the score of the effectors towards targets that are actually present in the host.</p>
<p>A host jump is now performed by creating a new host, with a different random list of targets. A sample of the pathogen
population is moved to the new host, and the new fitness is calculated.</p>
<p>Also, we keep track of the different strains and theur population dynamics in time.</p>

<h3>Discussion</h3>

<p></p> <!--some results and observations-->

<h3>Running simulations and getting your own results</h3>

<p>To run a simulation with the second model, download the codes <a href="https://github.com/Host-pathogen-evol/TSL-PROJECT-A/tree/master/SECOND_MODEL">here</a>.</p>
<p>You can have a full list of options running</p>
<code>$python hpmodel2.py -h</code>
<p>We suggest to seed the random number generator with the <code>-s</code> or <code>--seed</code> option, as the program is using a default seed.</p>
<p>For more information you can also rely on the docstring in the main code.</p>
<p>It is advisable to set the path to a new folder not to overwrite previous results, and because the program could produce several new files.</p>

<hr color=rgba(0,0,0,0.5)/>

<h3>Additional Scripts</h3>

<p>In the repository you will also find some scripts we wrote to plot the results. You may want to use these scripts or to
    write your own. To use the provided scripts you will need to download and install the <a href="http://matplotlib.org/users/installing.html">Matplotlib</a> library.<p>

<h4><bold>First Model</bold></h4> <!--this section need to be fixed, add parameters to give the script and what plot you get from it-->

<p>To plot the results of the simulation you will need to write in command line:
    <code>$python clusterV.py</code>
After getting the results (the script will save some plots AND some data files, that you will need for the next step), you can use the second script
    <code>$python comparison_plot.py</code>
You may also want to run the following script to get plots with the length distribution of effectors and TEs</p> <!--add code and check for dependencies-->

<h4><bold>Second Model</bold></h4>

<hr color=rgba(0,0,0,0.5)/>

<h3>Follow Up</h3>

<p>There's much more to be tested and implemented (this is a simple model, many features could be added in future).
It would be interesting to have some experimental results, as hard as it is to infer evolutionary patterns.
Another interesting thing still to be investigate would be the survival/evolution of a pathogen that, after some evolution time, jumps
back to its original host.</p>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/Host-pathogen-evol/TSL-PROJECT-A">The Genome Size Evolution Project</a> is maintained by <a href="https://github.com/calugo">Carlos Lugo</a> and <a href="https://github.com/aliceminotto">Alice Minotto</a>.</span>
        <span class="site-footer-owner">You can contact us at <a href="mailto:kupkasmale@gmail.com?subject=genome%20size%20evolution%20project">Carlos Lugo</a>.</span>
        <span class="site-footer-credits">This page was generated by <a href="https://pages.github.com">GitHub Pages</a> modifying the <a href="https://github.com/jasonlong/cayman-theme">Cayman theme</a> by <a href="https://twitter.com/jasonlong">Jason Long</a>.</span>
      </footer>

    </section>

    <script src="javascripts/jquery.min.js"></script>
    <script src="javascripts/jquery.waypoints.min.js"></script>
    <script src="javascripts/sticky.min.js"></script>
    <script>
        var sticky = new Waypoint.Sticky({
            element: $('#navigator')
        })
    </script>
  </body>
</html>
